pkg/sql/schemachanger/scbuild/internal/scbuildstmt/drop_index.go

```diff
@@ -270,15 +270,18 @@ func maybeDropDependentFKConstraints(
 	// dropDependentFKConstraint is a helper function that drops a dependent
 	// FK constraint with ID `fkConstraintID`.
 	dropDependentFKConstraint := func(fkConstraintID catid.ConstraintID) {
-		b.BackReferences(tableID).Filter(hasConstraintIDAttrFilter(fkConstraintID)).
+		b.BackReferences(tableID).Filter(containsDescIDFilter(tableID)).Filter(hasConstraintIDAttrFilter(fkConstraintID)).
 			ForEach(func(
 				current scpb.Status, target scpb.TargetStatus, e scpb.Element,
 			) {
 				b.Drop(e)
 			})
 	}
 
-	b.BackReferences(tableID).ForEach(func(
+	// Get foreign key back references referring to this table, previous we
+	// did not explicitly filter these out which could lead to us blocking
+	// the DROP INDEX or worse removing unrelated foreign keys.
```

Comment:
Do we want to put the history into the comment here? Maybe just say "Iterate over all FKs inbound to this table and decide whether we should drop them as a result of dropping this uniqueness-providing constraint"

Reply:
Good point. I'll simplify this to:

Iterate over all FKs inbound to this table and decide whether any other unique constraints will satisfy them if we were to drop the current unique constraint.